File: 242_valid_anagram/valid_anagram.go
package main

import "fmt"

type Solution struct {
}

func (solution *Solution) is_anagram(s, t string) bool {
	if len(s) != len(t) {
		return false
	}

	counter := make(map[rune]int)

	for _, ch := range s {
		counter[ch]++
	}

	for _, ch := range t {
		if counter[ch] == 0 {
			return false
		}
		counter[ch]--
	}

	return true

}

func main() {
	solution := Solution{}
	s := "anagram"
	t := "nagaram"

	result := solution.is_anagram(s, t)
	fmt.Println(result)
}

File: 242_valid_anagram/valid_anagram.py
class Solution:
    def valid_anagram(self, s: str, t: str) -> bool:  
        if len(s) != len(t):
            return False

        counter = {}

        for char in s:
            counter[char] = counter.get(char, 0) + 1

        for char in t:
            if char not in counter or counter[char] == 0:
                return False
            counter[char] -= 1

        return True

if __name__ == "__main__":
    solution = Solution()
    s = "anagram"
    t = "nagaram"

    result = solution.valid_anagram(s, t)
    print(result)